internal/conf/reposource/go_modules.go

```diff
@@ -27,11 +33,28 @@ func ParseGoModDependency(dependency string) (*GoDependency, error) {
 		mod.Version = dependency[i+1:]
 	}
 
-	err := module.Check(mod.Path, mod.Version)
+	var err error
+	if mod.Version != "" {
+		err = module.Check(mod.Path, mod.Version)
+	} else {
+		err = module.CheckPath(mod.Path)
+	}
+
 	if err != nil {
 		return nil, err
 	}
-	return &GoDependency{mod: mod}, nil
+
+	return &GoDependency{Module: mod}, nil
+}
+
+// ParseGoDependencyFromRepoName is a convenience function to parse a repo name in a
+// 'go/mod(@version)?' format into a GoDependency.
```

Comment:
```suggestion
// 'go/<name>(@<version>)?' format into a GoDependency.
```

For consistency with spelling earlier.